### ISSUE TITLE
Questions about "decoder_nll"

### ISSUE THREAD

**Issue by scvready123** [Shark-NLP/DiffuSeq#27]
Hi,thanks for your implementation.Your code is very clean and clear.But here I've got one problem.
That is in diffuseq/gaussian_diffusion.py,from line 632 to line 636.Here we got 2 nlls:
decoder_nll = self._token_discrete_loss(x_start, get_logits, input_ids_x) # embedding regularization
terms["nll"] = self._token_discrete_loss(model_out_x_start, get_logits, input_ids_x, mask=input_ids_mask, truncate=True, t=t)
and I see you use the first one to compute the losses:
terms["loss"] = terms["mse"] + decoder_nll + tT_loss
So I wonder why you use the decoder_nll rather than terms["nll"].I think decoder_nll doesn't use the transformer part of the model,can it benefit the training process?
I would appreciate it if you could answer this question.

**Comment by summmeer** [comment]
Hi,
Plz refer to #25

**Comment by scvready123** [comment]
Thank you very much!But I still don't understand what does "nll loss is for tracing the generation quality" mean as you said in #25.
Can you explain it?

**Comment by summmeer** [comment]
it is a metric to trace the nll of reconstruct $x_0$, and through the training nll / eval nll, we can tell whether the training process is normal or not.
